demisto_sdk/commands/generate_docs/generate_integration_doc.py

```diff
@@ -31,6 +42,402 @@
+                else:
+                    old_cmd_end_line = doc_text_lines.index(old_command_section[-2])
+
+                doc_text_lines[
+                    old_cmd_start_line : old_cmd_end_line + 1
+                ] = new_command_section
+
+                self.output_doc = "\n".join(doc_text_lines)
+            except (ValueError, IndexError) as e:
+                error = f"Unable to replace '{modified_command}' section in README: {str(e)}"
+                self.update_errors.append(error)
+
+    def _get_sections_to_update(self) -> Tuple[bool, List[str], List[str]]:
+
+        return (
+            self.integration_diff.is_configuration_different(),
+            self.integration_diff.get_modified_commands(),
+            self.integration_diff.get_added_commands(),
+        )
+
+    def update_docs(self) -> Tuple[str, List[str]]:
+        """
+        Helper function that updates the integration documentation and returns the
+        raw README text.
+        """
+
+        (
+            update_configuration,
+            modified_commands,
+            added_commands,
+        ) = self._get_sections_to_update()
+
+        # In case the configuration section has changed
+        # we want to replace the section with the new
+        if update_configuration:
+            logger.info("\t\u2699 Integration configuration has changed, updating...")
+
+            self._update_conf_section()
+
+        # Handle changed command arguments
+        # We iterate over every modified command and subtitute
+        # the section with the new one
+        if modified_commands:
+            logger.info(
+                f"\t\u2699 Integration commands {','.join(modified_commands)} have changed, replacing the old section with the new one..."
+            )
+
+            self._update_commands_section()
+
+        # Handle new commands
+        # we append them to the README.
+        if added_commands:
+            self.output_doc += "\n"
+
+            for cmd in added_commands:
+                logger.info(f"\t\u2699 Generating docs for command `{cmd}`...")
+                (
+                    command_section,
+                    generate_command_section_errors,
+                ) = generate_commands_section(
+                    yaml_data=self.integration_diff.new_yaml_data,
+                    example_dict=self.example_dict,
+                    command_permissions_dict=self.command_permissions_dict,
+                    command=cmd,
+                )
+
+                if generate_command_section_errors:
+                    logger.error(
+                        f"\t\u26A0 Generating section for command '{cmd}' resulted in {len(generate_command_section_errors)} error(s)"
+                    )
+                    self.update_errors.extend(generate_command_section_errors)
+
+                command_section_str = "\n".join(command_section)
+                self.output_doc, append_cmd_errs = append_or_replace_command_in_docs(
```

Comment:
`append_or_replace_command_in_docs` attempts to find the an input command in a section. If it doesn't find it, it appends it to the end. If it does find it, it's supposed to replace it within the input doc text. So I attempted to use this instead of the `IntegrationDocUpdateManager::_update_commands_section` method. However, this resulted in the commands sections needed to be updated not working as expected. Therefore, I will continue using  `IntegrationDocUpdateManager::_update_commands_section`.